test/tools/calculateChanges.test.ts

```diff
@@ -0,0 +1,338 @@
+import { expect } from 'chai';
+import { calculateChanges, processChangedObjectFields } from '../../src/tools/calculateChanges';
+import DefaultHandler from '../../src/tools/auth0/handlers/default';
+import configFactory from '../../src/configFactory';
+
+describe('#utils calcChanges', () => {
+    class MockHandler extends DefaultHandler {
+        constructor(settings = {}) {
+            const config = configFactory();
+            // eslint-disable-next-line no-restricted-syntax
+            for (const key of Object.keys(settings)) {
+                config.setValue(key, settings[key]);
+            }
+
+            super({
+                config,
+                type: 'mock'
+            });
+        }
+    }
+
+    const mockHandler = new MockHandler();
+
+    const mockHandlerWithObjectFields = (() => {
+        const handler = new MockHandler();
+        handler.objectFields = ['metadata'];
+        return handler;
+    })();
+
+    it('should calc create', () => {
+        const existing = [{ name: 'Name1', id: 'id1' }];
+        const assets = [
+            { name: 'Name1', id: 'id3' },
+            { name: 'Create1', id: 'Create1' }
+        ];
+
+        const { create } = calculateChanges({ handler: mockHandler, assets, existing, identifiers: ['id', 'name'], allowDelete: false });
+
+        expect(create).to.have.length(1);
+        expect(create).to.deep.include({ name: 'Create1', id: 'Create1' });
+    });
+
+    it('should calc create grouped identifiers', () => {
+        const existing = [{ client_id: 'client1', audience: 'audience1', id: 'id1' }];
+        const assets = [
+            { client_id: 'client1', audience: 'audience1', id: 'id3' },
+            { client_id: 'create1', audience: 'create1', id: 'create1' }
+        ];
+
+        const { create } = calculateChanges({
+            handler: mockHandler,
+            assets,
+            existing,
+            allowDelete: false,
+            //@ts-ignore need to investigate why this "grouping" exists
```

Comment:
This is one oddity that still needs to be understood, why we're passing in an array of mixed types. This test will pass if the nested array is broken down; not really sure what it's doing by looking at the code. 

Nevertheless, will not be refactoring, just something to point out.